hamlet/views.py

```diff
@@ -116,11 +116,18 @@ def ml_subject_assistant_list(request, project_id):
         ml_subject_assistant_exports = []
 
         for subject_set in SubjectSet.where(project_id=project_id):
+            data_export = MLSubjectAssistantExport.objects.filter(
+                subject_set_id=subject_set.id
+            ).order_by('-created')
+            
+            external_web_app_url = settings.SUBJECT_ASSISTANT_EXTERNAL_URL 
+            if list(data_export)[0].ml_task_id:
+                external_web_app_url = external_web_app_url + str(list(data_export)[0].ml_task_id)
```

Comment:
It looks like you can use the `add` template filter to join strings. It does try to convert the strings to integers first to sum them, but it should be safe here since the first string is always going to be a URL.

https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#add

```
{{ external_web_app_url|add:ml_task_id }}
```

Even so, it might be clearer to join them in the view like you have done. I'll let you decide.

Reply:
Thanks Adam, that really helped me figure out templates. So what I found is that...

```
Yo check 'dis out: {{ url_as_string|add:id_as_number }}
``` 

...results in... `Yo check 'dis out:` (blank)

`|add`-ing string with string works fine, `|add`-ing numbers with numbers too, but for concatenating strings with numbers, [the solution](https://stackoverflow.com/questions/18519109/how-to-concatenate-a-string-to-a-number-inside-a-template-tag-in-django) I found that worked is...

```
{% with id_as_number|stringformat:"s" as id_as_string %}
  Yo check 'dis out: {{ url_as_string|add:id_as_string }}
{% endwith %}
```

That's a bit less readable than just formatting `external_web_app_url` in `views.py` though, so I'll keep that as-is. (And I have to write down this `|stringformat` `|add` pattern somewhere else so I don't forget it.)